crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -1,45 +1,101 @@
+        (currency_unit, currency, amount, item): (&api::CurrencyUnit, enums::Currency, i64, T),
     ) -> Result<Self, Self::Error> {
-        //Todo :  use utils to convert the amount to the type of amount that a connector accepts
+        let amount = utils::get_amount_as_string(currency_unit, amount, currency)?;
         Ok(Self {
             amount,
             router_data: item,
         })
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub mod auth_headers {
+    pub const API_VERSION: &str = "api-version";
+}
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct MifinityConnectorMetadataObject {
+    pub brand_id: Secret<String>,
+}
+
+impl TryFrom<&Option<pii::SecretSerdeValue>> for MifinityConnectorMetadataObject {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(meta_data: &Option<pii::SecretSerdeValue>) -> Result<Self, Self::Error> {
+        let metadata: Self = utils::to_connector_meta_from_secret::<Self>(meta_data.clone())
+            .change_context(errors::ConnectorError::InvalidConnectorConfig {
+                config: "merchant_connector_account.metadata",
+            })?;
+        Ok(metadata)
+    }
+}
+
+fn get_brand_id(
+    connector_metadata: &Option<pii::SecretSerdeValue>,
+) -> CustomResult<String, errors::ConnectorError> {
+    let mifinity_metadata = MifinityConnectorMetadataObject::try_from(connector_metadata)?;
+    let brand_id = mifinity_metadata.brand_id.clone().expose();
+    Ok(brand_id)
+}
+
+#[derive(Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
 pub struct MifinityPaymentsRequest {
-    amount: i64,
-    card: MifinityCard,
+    money: Money,
+    client: MifinityClient,
+    address: MifinityAddress,
+    validation_key: String,
+    client_reference: common_utils::id_type::CustomerId,
+    trace_id: String,
+    description: String,
+    destination_account_number: Secret<String>,
+    brand_id: Secret<String>,
+    return_url: String,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct MifinityCard {
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+#[derive(Debug, Serialize, PartialEq)]
+pub struct Money {
+    amount: String,
+    currency: String,
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct MifinityClient {
+    first_name: Secret<String>,
+    last_name: Secret<String>,
+    phone: Secret<String>,
+    dialing_code: String,
+    nationality: api_models::enums::CountryAlpha2,
+    email_address: Email,
+    dob: Secret<Date>,
+}
+
+#[derive(Default, Debug, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct MifinityAddress {
+    address_line1: Secret<String>,
+    country_code: api_models::enums::CountryAlpha2,
+    city: String,
```

Comment:
@SamraatBansal Should we consider making this a `Secret<String>`?

Reply:
City and Country are not sensitive info according to pii guidelines.
Core also does not consider it as secret so same is translated here